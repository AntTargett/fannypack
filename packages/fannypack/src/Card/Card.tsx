import * as React from 'react';
import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook, useUniqueId } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalCardProps = {
  kind?: 'border' | 'shadow';
  standalone?: boolean;
  title?: string | React.ReactElement<any>;
  footer?: string | React.ReactElement<any>;
  headerAddon?: React.ReactElement<any>;
};
export type CardProps = BoxProps & LocalCardProps;

export type CardContextOptions = CardProps & { descriptionId?: string; titleId?: string; themeKey?: string };
export const CardContext = React.createContext<CardContextOptions>({});

const useProps = createHook<CardProps>(
  (props, themeKey) => {
    const { footer, headerAddon, standalone, title, kind, ...restProps } = props;
    const boxProps = Box.useProps({
      altitude: kind === 'shadow' ? '100' : null,
      border: kind === 'border' ? 'default' : null,
      ...restProps
    });

    const className = useClassName({
      style: styles.Card,
      styleProps: props,
      themeKey,
      prevClassName: boxProps.className
    });

    const titleId = useUniqueId('cardTitle');
    const descriptionId = useUniqueId('cardDescription');

    const context = React.useMemo(() => ({ descriptionId, titleId, ...props }), [descriptionId, props, titleId]);

    const children = (
      <CardContext.Provider value={context}>
        {standalone ? (
          props.children
        ) : (
          <React.Fragment>
            {title && (
              <CardHeader>
                <CardTitle>{title}</CardTitle>
                {headerAddon && <Box>{headerAddon}</Box>}
              </CardHeader>
            )}
            <CardContent>{props.children}</CardContent>
            {footer && <CardFooter>{footer}</CardFooter>}
          </React.Fragment>
        )}
      </CardContext.Provider>
    );

    return {
      'aria-describedby': props.children ? descriptionId : undefined,
      'aria-labelledby': props.title ? titleId : undefined,
      ...boxProps,
      className,
      children
    };
  },
  {
    defaultProps: {
      kind: 'shadow'
    },
    themeKey: 'Card'
  }
);

export const Card = createComponent<CardProps>(
  props => {
    const cardProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: cardProps });
  },
  {
    attach: {
      useProps
    },
    themeKey: 'Card'
  }
);

/////////////////////////////////////

export type LocalCardContentProps = {};
export type CardContentProps = BoxProps & LocalCardContentProps;

export function CardContent(props: CardContentProps) {
  const { children, ...restProps } = props;
  const context = React.useContext(CardContext);

  const cardContentClassName = useClassName({
    style: styles.CardContent,
    styleProps: { ...context, ...props },
    themeKey: `${context.themeKey || 'Card'}.Content`
  });

  return (
    <Box className={cardContentClassName} id={context.descriptionId} {...restProps}>
      {children}
    </Box>
  );
}

/////////////////////////////////////

export type LocalCardTitleProps = {};
export type CardTitleProps = BoxProps & LocalCardTitleProps;

export function CardTitle(props: CardTitleProps) {
  const { children, ...restProps } = props;
  const context = React.useContext(CardContext);

  const cardTitleClassName = useClassName({
    style: styles.CardTitle,
    styleProps: { ...context, ...props },
    themeKey: `${context.themeKey || 'Card'}.Title`
  });

  return (
    <Box className={cardTitleClassName} id={context.titleId} {...restProps}>
      {children}
    </Box>
  );
}

/////////////////////////////////////

export type LocalCardHeaderProps = {};
export type CardHeaderProps = BoxProps & LocalCardHeaderProps;

export function CardHeader(props: CardHeaderProps) {
  const { children, ...restProps } = props;
  const context = React.useContext(CardContext);

  const cardHeaderClassName = useClassName({
    style: styles.CardHeader,
    styleProps: { ...context, ...props },
    themeKey: `${context.themeKey || 'Card'}.Header`
  });

  return (
    <Box className={cardHeaderClassName} {...restProps}>
      {children}
    </Box>
  );
}

/////////////////////////////////////

export type LocalCardFooterProps = {};
export type CardFooterProps = BoxProps & LocalCardFooterProps;

export function CardFooter(props: CardFooterProps) {
  const { children, ...restProps } = props;
  const context = React.useContext(CardContext);

  const cardFooterClassName = useClassName({
    style: styles.CardFooter,
    styleProps: { ...context, ...props },
    themeKey: `${context.themeKey || 'Card'}.Footer`
  });

  return (
    <Box className={cardFooterClassName} {...restProps}>
      {children}
    </Box>
  );
}
