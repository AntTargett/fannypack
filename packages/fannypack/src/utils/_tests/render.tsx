import * as React from 'react';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (Component: any, { theme = {} } = {}) => render(<Provider theme={theme}>{Component}</Provider>);
