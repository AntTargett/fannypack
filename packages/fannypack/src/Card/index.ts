import Card from './Card';

export default Card;

export * from './Card';
export { default as Card } from './Card';
export * from './CardCard';
export { default as CardCard } from './CardCard';
export * from './CardContent';
export { default as CardContent } from './CardContent';
export * from './CardFooter';
export { default as CardFooter } from './CardFooter';
export * from './CardHeader';
export { default as CardHeader } from './CardHeader';
export * from './CardTitle';
export { default as CardTitle } from './CardTitle';
