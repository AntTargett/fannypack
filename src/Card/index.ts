import Card from './Card';
import * as cardStyled from './styled';

export * from './Card';
export * from './CardCard';
export * from './CardContent';
export * from './CardFooter';
export * from './CardHeader';
export * from './CardTitle';
export { cardStyled };
export default Card;
