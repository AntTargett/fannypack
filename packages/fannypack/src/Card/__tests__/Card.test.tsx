import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Card } from '../Card';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Card>Hello world</Card>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Card color="primary">Hello world</Card>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it('should render correctly', () => {
      const { container } = render(<Card use="div">Hello world</Card>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Card props', () => {
      const { result } = renderHook(() => Card.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(<Card>{CardProps => <div {...CardProps}>Hello world</div>}</Card>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Card.base should render correctly', () => {
    const { container } = render(<Card overrides={{ Card: { base: { backgroundColor: 'red' } } }}>hello world</Card>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Card.base should render correctly', () => {
    const { container } = render(<Card>hello world</Card>, {
      theme: { Card: { base: { backgroundColor: 'red' } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Card>hello world</Card>, {
      theme: { Card: { defaultProps: { className: 'test' } } }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
