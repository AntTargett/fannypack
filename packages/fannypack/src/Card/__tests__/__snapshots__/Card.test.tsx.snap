// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
</div>
`;

exports[`composition hook should return with Card props 1`] = `
Object {
  "aria-describedby": undefined,
  "aria-labelledby": undefined,
  "children": <Context.Provider
    value={
      Object {
        "descriptionId": "",
        "titleId": "",
        "type": "shadow",
      }
    }
  >
    <React.Fragment>
      <CardContent />
    </React.Fragment>
  </Context.Provider>,
  "className": "css-gziovy css-1pwrqjj css-zp6o43 fp-Box fp-Card",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-2 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  padding: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Card"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 test fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`defaultProps should render correctly for title 1`] = `
.emotion-11 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  padding: 1.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

<div
  aria-describedby="cardDescription"
  aria-labelledby="cardTitle"
  class="emotion-9 emotion-0 emotion-11 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-5 fp-Card.Header fp-Box"
  >
    <div
      class="emotion-0 emotion-1 emotion-2 fp-Card.Title fp-Box"
      id="cardTitle"
    >
      This is a title
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-8 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`defaultProps should render correctly for type 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

.emotion-5.emotion-5.emotion-5 {
  border: 1px solid #e6e6e6;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`overrides Card.Content.base should render correctly 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

.emotion-2 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`overrides Card.Footer.base should render correctly 1`] = `
.emotion-10 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-8 {
  padding: 1.5rem;
}

.emotion-7 {
  margin-top: 1.5rem;
}

.emotion-7 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  class="emotion-8 emotion-0 emotion-10 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
  <div
    class="emotion-0 emotion-1 emotion-7 fp-Card.Footer fp-Box"
  >
    <div
      class="emotion-0 emotion-1 fp-Box"
    >
      This is a footer
    </div>
  </div>
</div>
`;

exports[`overrides Card.Header.base should render correctly 1`] = `
.emotion-11 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  padding: 1.5rem;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-5 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  aria-labelledby="cardTitle"
  class="emotion-9 emotion-0 emotion-11 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-5 fp-Card.Header fp-Box"
  >
    <div
      class="emotion-0 emotion-1 emotion-2 fp-Card.Title fp-Box"
      id="cardTitle"
    >
      This is a title
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-8 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`overrides Card.Title.base should render correctly 1`] = `
.emotion-11 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  padding: 1.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-2 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  aria-labelledby="cardTitle"
  class="emotion-9 emotion-0 emotion-11 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-5 fp-Card.Header fp-Box"
  >
    <div
      class="emotion-0 emotion-1 emotion-2 fp-Card.Title fp-Box"
      id="cardTitle"
    >
      This is a title
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-8 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`overrides Card.base should render correctly 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

.emotion-3 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly for a footer 1`] = `
.emotion-10 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-8 {
  padding: 1.5rem;
}

.emotion-7 {
  margin-top: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-8 emotion-0 emotion-10 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
  <div
    class="emotion-0 emotion-1 emotion-7 fp-Card.Footer fp-Box"
  >
    <div
      class="emotion-0 emotion-1 fp-Box"
    >
      This is a footer
    </div>
  </div>
</div>
`;

exports[`props should render correctly for a title 1`] = `
.emotion-11 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  padding: 1.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

<div
  aria-describedby="cardDescription"
  aria-labelledby="cardTitle"
  class="emotion-9 emotion-0 emotion-11 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-5 fp-Card.Header fp-Box"
  >
    <div
      class="emotion-0 emotion-1 emotion-2 fp-Card.Title fp-Box"
      id="cardTitle"
    >
      This is a title
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-8 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly for headerAddon 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly for standalone 1`] = `
.emotion-14 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-12 {
  padding: 1.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-11 {
  margin-top: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-12 emotion-0 emotion-14 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-5 fp-Card.Header fp-Box"
  >
    <div
      class="emotion-0 emotion-1 emotion-2 fp-Card.Title fp-Box"
      id="cardTitle"
    >
      This is a title
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-8 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse diam ipsum, cursus id placerat congue, ultrices eget lectus. Duis posuere, lacus sed tristique commodo, sapien turpis mollis nunc, vestibulum consectetur lectus augue sit amet justo.
  </div>
  <div
    class="emotion-0 emotion-1 emotion-11 fp-Card.Footer fp-Box"
  >
    This is a footer
  </div>
</div>
`;

exports[`props should render correctly for type border 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

.emotion-5.emotion-5.emotion-5 {
  border: 1px solid #e6e6e6;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly for type shadow 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

.emotion-5.emotion-5.emotion-5 {
  color: #3722ed;
}

.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    Hello world
  </div>
</div>
`;

exports[`theming Card.Content.base should render correctly 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

.emotion-2 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`theming Card.Footer.base should render correctly 1`] = `
.emotion-10 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-8 {
  padding: 1.5rem;
}

.emotion-7 {
  margin-top: 1.5rem;
}

.emotion-7 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  class="emotion-8 emotion-0 emotion-10 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
  <div
    class="emotion-0 emotion-1 emotion-7 fp-Card.Footer fp-Box"
  >
    <div
      class="emotion-0 emotion-1 fp-Box"
    >
      This is a footer
    </div>
  </div>
</div>
`;

exports[`theming Card.Header.base should render correctly 1`] = `
.emotion-11 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  padding: 1.5rem;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-5 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  aria-labelledby="cardTitle"
  class="emotion-9 emotion-0 emotion-11 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-5 fp-Card.Header fp-Box"
  >
    <div
      class="emotion-0 emotion-1 emotion-2 fp-Card.Title fp-Box"
      id="cardTitle"
    >
      This is a title
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-8 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`theming Card.Title.base should render correctly 1`] = `
.emotion-11 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-9 {
  padding: 1.5rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-2 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  aria-labelledby="cardTitle"
  class="emotion-9 emotion-0 emotion-11 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-5 fp-Card.Header fp-Box"
  >
    <div
      class="emotion-0 emotion-1 emotion-2 fp-Card.Title fp-Box"
      id="cardTitle"
    >
      This is a title
    </div>
  </div>
  <div
    class="emotion-0 emotion-1 emotion-8 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;

exports[`theming Card.base should render correctly 1`] = `
.emotion-5 {
  box-shadow: 0px 2px 4px 0px rgba(0,0,0,0.1),0px 0px 2px 1px rgba(0,0,0,0.02);
}

.emotion-0 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-0:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  padding: 1.5rem;
}

.emotion-3 {
  background-color: red;
}

<div
  aria-describedby="cardDescription"
  class="emotion-3 emotion-0 emotion-5 fp-Box fp-Card"
>
  <div
    class="emotion-0 emotion-1 emotion-2 fp-Card.Content fp-Box"
    id="cardDescription"
  >
    hello world
  </div>
</div>
`;
